### DISCUSSION TITLE
Issue with Token Import Approval after WalletConnect Connection

### DISCUSSION THREAD

**Discussion by ArmmGh** [WalletConnect/walletconnect-monorepo#4024] [Developer Support]
Hello,

Here's a brief overview of the problem:

1. User initiates token import.
2. User connects via WalletConnect.
3. User approves connection in MetaMask.
4. After connection approval, our code triggers watchAsset for token import approval.
5. However, the user is not redirected back to MetaMask for the second approval when returning to the website.

I suspect that the approval flow might be impacted by the timing or event handling during the WalletConnect connection process. Despite attempting to trigger watchAsset with a delay of 10 seconds using a timeout, the issue persists.

Interestingly, if I manually click on the button triggering the import token function while being connected via walletConnect, it successfully redirects me to MetaMask for token import approval.

Could you provide insights or recommendations in scenarios where manual triggering succeeds but automatic triggering encounters issues?

Are there any specific considerations or best practices in viem for handling sequential approval steps after a WalletConnect connection?

Any guidance or insights you can provide would be greatly appreciated.

**Comment by glitch-txs** [comment]
hi, what libaries and versions are you using? is also on iOS? because if so then it's related to the way they handle security (you need a direct user action to redirect from a website)

**Reply by ArmmGh** [reply]
Hi, 
I use 
- wagmi + viem latest versions
- tested on IOS | Android
I noticed that if I connect via walletConnect and accept network switch to the right chain, then my walletClient (which I need to send watchAsset request) is null and is not updating even though on MM I'm connected on the right chain, but after I refresh website it connects normally.
